homeassistant/components/sensor/jewish_calendar.py

```diff
@@ -0,0 +1,141 @@
+    'gra_end_shma': ['Latest time for Shm"a GR"A', 'mdi:calendar-clock'],
+    'mga_end_shma': ['Latest time for Shm"a MG"A', 'mdi:calendar-clock'],
+    'plag_mincha': ['Plag Hamincha', 'mdi:weather-sunset-down'],
+    'first_stars': ['T\'set Hakochavim', 'mdi:weather-night'],
+}
+
+CONF_DIASPORA = 'diaspora'
+CONF_LANGUAGE = 'language'
+CONF_SENSORS = 'sensors'
+
+DEFAULT_NAME = 'Jewish Calendar'
+DEFAULT_LATITUDE = 31.778
+DEFAULT_LONGITUDE = 35.235
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_DIASPORA, default=False): cv.boolean,
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_LANGUAGE, default='english'):
+        vol.In(['hebrew', 'english']),
+    vol.Optional(CONF_SENSORS, default=['date']):
+        vol.All(cv.ensure_list, vol.Length(min=1), [vol.In(SENSOR_TYPES)]),
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the Jewish calendar sensor platform."""
+    language = config.get(CONF_LANGUAGE)
+    name = config.get(CONF_NAME)
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    diaspora = config.get(CONF_DIASPORA)
+
+    if None in (latitude, longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return False
+
+    dev = []
+    for sensor_type in config[CONF_SENSORS]:
+        dev.append(JewishCalSensor(
+            name, language, sensor_type, latitude, longitude, diaspora))
+    async_add_entities(dev)
+
+
+class JewishCalSensor(Entity):
+    """Representation of an Jewish calendar sensor."""
+
+    def __init__(self, name, language, sensor_type,
+                 latitude=DEFAULT_LATITUDE, longitude=DEFAULT_LONGITUDE,
+                 diaspora=False):
+        """Initialize the Jewish calendar sensor."""
+        self.client_name = name
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self.type = sensor_type
+        self._date = dt_util.now().date()
+        self._hebrew = (language == 'hebrew')
+        self._state = None
+        self.latitude = latitude
+        self.longitude = longitude
+        self.diaspora = diaspora
+        _LOGGER.debug("Sensor %s initialized with date %s",
+                      self.type, self._date)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def icon(self):
+        """Icon to display in the front end."""
+        return SENSOR_TYPES[self.type][1]
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    async def async_update(self):
+        """Update the state of the sensor."""
+        import hdate
+
+        self._date = dt_util.now().date()
```

Comment:
There is no reason to store this as an instance variable, it's only used in this method.